src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,175 @@
     }
 
     /**
      * Given the date and the number of weeks to add, return the changed date.
      */
     public LocalDate addWeeks(LocalDate localDate, Integer numberOfWeeks) {
-        return LocalDate.now();
+        return localDate.plusWeeks(numberOfWeeks);
     }
 
     /**
      * Given a random `someDate` date, return one of the following Strings:
-     *     - "`someDate` is after `currentDate`"
-     *                  if `someDate` is in the future relating to the `current date`;
-     *     - "`someDate` is before `currentDate`"
-     *                  if `someDate` is in the past relating to the `current date`;
-     *     - "`someDate` is today"
-     *                  if `someDate` is today;
+     * - "`someDate` is after `currentDate`"
+     * if `someDate` is in the future relating to the `current date`;
+     * - "`someDate` is before `currentDate`"
+     * if `someDate` is in the past relating to the `current date`;
+     * - "`someDate` is today"
+     * if `someDate` is today;
      */
     public String beforeOrAfter(LocalDate someDate) {
-        return someDate + "is today";
+        return someDate.isEqual(NOW)
+                ? someDate + " is today"
+                : someDate.isAfter(NOW)
+                ? someDate + " is after " + NOW
+                : someDate + " is before " + NOW;
     }
 
     /**
      * Given a String representation of a date and some timezone,
      * return LocalDateTime in this timezone.
      */
     public LocalDateTime getDateInSpecificTimeZone(String dateInString, String zone) {
-        return LocalDateTime.now();
+        return Instant.parse(dateInString)
+                .atZone(ZoneId.of(zone)).toLocalDateTime();
     }
 
     /**
      * Given some LocalDateTime, return an OffsetDateTime with the local time offset applied
      * (`+02:00` for Ukraine).
-     *
+     * <p>
      * Example: we receive a LocalDateTime with a value `2019-09-06T13:17`.
-     *          We should return the OffsetDateTime with a value `2019-09-06T13:17+02:00`,
-     *          where `+02:00` is the offset for our local timezone.
-     *
+     * We should return the OffsetDateTime with a value `2019-09-06T13:17+02:00`,
+     * where `+02:00` is the offset for our local timezone.
+     * <p>
      * OffsetDateTime is recommended to use for storing date values in a database.
      */
     public OffsetDateTime offsetDateTime(LocalDateTime localTime) {
-        return OffsetDateTime.now();
+        return OffsetDateTime.of(localTime, ZoneOffset.of(UKRAINE_TIMEZONE));
     }
 
     /**
      * Given a String formatted as `yyyymmdd`,
      * return Optional of this date as a LocalDate.
      */
     public Optional<LocalDate> parseDate(String date) {
-        return Optional.empty();
+        return Integer.parseInt(date.substring(DAY_START, DAY_END)) <= NUMBER_OF_DAYS
+                ? Optional.of(LocalDate.parse(date, DateTimeFormatter.ofPattern(DATE_FORMAT)))
```

Comment:
you can get rid of parseInt and use try catch DateTimeParseException the reason is the same as above

Reply:
wrapped into `try-catch` block and refactored for a one-liner to return formatted `LocalDate`